pg_search/src/api/operator/searchconfig.rs

```diff
@@ -42,8 +43,19 @@ pub fn search_with_search_config(
         let search_config: SearchConfig = serde_json::from_value(search_config_json.clone())
             .expect("could not parse search config");
 
+        let index_name = &search_config.index_name;
+        let index_oid = &search_config.index_oid;
+        let database_oid = crate::MyDatabaseId();
+        let oid_query =
+            format!("SELECT relfilenode FROM pg_class WHERE oid = {index_oid} AND relkind = 'i'",);
+        let relfilenode = match Spi::get_one::<pg_sys::Oid>(&oid_query) {
+            Ok(Some(relfilenode)) => relfilenode,
+            Ok(None) => panic!("no relfilenode for index '{index_name}' in snippet"),
+            Err(err) => panic!("error looking up index '{index_name}': {err}"),
+        };
+
```

Comment:
We don't need to go through SPI to get the relfilenode.  We can just open a `PgRelation` with the `index_oid` and get it through the struct.

And I think we should probably have a function in `utils.rs` that does that, rather than inlining that logic at all the call sites that need it.